tmva/sofie_parsers/src/RModelParser_ONNX.cxx

```diff
@@ -1308,16 +1308,6 @@ std::unique_ptr<ROperator> make_ROperator_Concat(const onnx::NodeProto &nodeprot
 
 std::unique_ptr<ROperator> make_ROperator_Shape(const onnx::NodeProto& nodeproto, const onnx::GraphProto& /*graphproto */, std::unordered_map<std::string, ETensorType>& tensor_type){
 
-   ETensorType input_type;
-
-   auto input_name =  nodeproto.input(0);
-   auto it = tensor_type.find(input_name);
-   if (it != tensor_type.end()){
-      input_type = it->second;
-   }else{
-      throw std::runtime_error("TMVA::SOFIE ONNX Parser Shape op has input tensor" + input_name + " but its type is not yet registered");
```

Comment:
I would not remove this part checking if the input tensor is registered already. 
This is needed to ensure the validity of the model and its consistency. We don't need instead the check later if the input type is equal to float and that is correct to remove

Reply:
You can remove instead line `1316`, 
```
input_type = it->second;
```
and just keep the error condition:
```
if (it == tensor_type.end()){
throw std::runtime_error("TMVA::SOFIE ONNX Parser Shape op has input tensor" + input_name + " but its type is not yet registered");
}
```